src/main/scala/com/actian/spark_vector/writer/DataStreamPartitionAssignment.scala

```diff
@@ -99,22 +100,35 @@ sealed trait BipartiteAssignment extends Logging with Profiling {
     foundPath
   }
 
+  private def logGraph: Unit = if (log.isTraceEnabled) {
```

Comment:
prefix with `@inline` or use `private final` and perhaps the compiler will inline the method if/where it can

[1]  http://alvinalexander.com/bookmarks/scala/scala-inline-final-example [inline def]
[2] https://gist.github.com/rxin/37554d9f3e2e93220884 [val getting inlined]